ext/pdo/tests/pdo_dsn_containing_credentials.phpt

```diff
@@ -0,0 +1,51 @@
+--TEST--
+PDO Common: Pass credentials in dsn instead of constructor params
+--SKIPIF--
+<?php
+if (!extension_loaded('pdo')) die('skip');
+$dir = getenv('REDIR_TEST_DIR');
+if (false == $dir) die('skip no driver');
+
+$driver = substr(getenv('PDOTEST_DSN'), 0, strpos(getenv('PDOTEST_DSN'), ':'));
+if (!in_array($driver, array('mssql','sybase','dblib','firebird','mysql','oci')))
```

Comment:
I'm just thinking out loud here... Shouldn't we then also enable the pgsql driver here?
```suggestion
if (!in_array($driver, array('mssql','sybase','dblib','firebird','mysql','oci','pgsql')))
```

Reply:
[Postgres already supports](https://www.php.net/manual/en/ref.pdo-pgsql.connection.php) putting credentials in the dsn - which makes its behavior differ from this (intentional b/c friendly change) - namely it considers the dsn as higher-prio than the constructor-arguments. This patch does it the other way around

Reply:
just to add - I wouldn't mind doing it the other way around for consistency. It would mean backward-breakage for people putting nonsensical credentials in the dsn, expecting the constructor arguments to take precendence

Reply:
I'm not sure that would be considered a BC break so probably mimicking the existing behaviour of Postgres makes more sense to me.

Reply:
Thanks for all these explanations... I see.

If mimicking the pgsql, the BC break for this is not that major to cause any issues in most of the apps. These would be really edge cases actually. It seems like a BC break technically speaking more or less. Yet, the code with such an edge case included in the DSN string wouldn't work the same as it would be on PHP 7.3.

Now, the case is that PHP-7.4 shouldn't contain BC breaks (although there are lots of them already implemented). We can break it but by having an RFC process. Or we break it in the PHP 8 (the semantic versioning way)... PHP unfortunately doesn't have standards here yet what is semver.

I'm in favour of breaking it also because having PDO constructor behaviour for different drivers here would be more confusing then...

@KalleZ what are your thoughts here?

Reply:
And also this then: thinking from the PDO end-user usage: Is it really more logical that DSN overrides the username and password? Or should be the otherway around and pgsql driver behavior changed?

Reply:
So, also another solution maybe here?
- leave the PR as it is and merge in the PHP-7.4
- deprecate the precedence for the pgsql (username and password passed as PDO constructor arguments override the ones in the DSN)
- change the precedence for the pgsql in PHP 8.0+

Reply:
@petk I agree with your last suggestion - leave this PR as it is and update pg to confirm in 8

Reply:
I like your approach @petk, if @SjonHortensius is fine with that then lets go about that. I guess the ext/pdo_pgsql changes can come in a separate PR.